user-info-provider/src/main/java/de/intension/mapper/oidc/RoleBasedUserInfoProviderMapper.java

```diff
@@ -0,0 +1,135 @@
+package de.intension.mapper.oidc;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.keycloak.models.ClientSessionContext;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ProtocolMapperContainerModel;
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.protocol.ProtocolMapperConfigException;
+import org.keycloak.protocol.ProtocolMapperUtils;
+import org.keycloak.protocol.oidc.mappers.AbstractOIDCProtocolMapper;
+import org.keycloak.protocol.oidc.mappers.OIDCAccessTokenMapper;
+import org.keycloak.protocol.oidc.mappers.OIDCAttributeMapperHelper;
+import org.keycloak.protocol.oidc.mappers.OIDCIDTokenMapper;
+import org.keycloak.protocol.oidc.mappers.UserInfoTokenMapper;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.IDToken;
+
+import de.intension.mapper.user.UserInfoHelper;
+
+/**
+ * Provides a custom implementation based on the user property protocol mapper. This custom logic
+ * includes the checking of roles in user attributes and then adding the user property to the claim.
+ */
+public class RoleBasedUserInfoProviderMapper extends AbstractOIDCProtocolMapper
+    implements OIDCAccessTokenMapper, OIDCIDTokenMapper, UserInfoTokenMapper
+{
+
+    public static final String                        ROLES_ATTRIBUTE_NAME     = "professionalRoles";
```

Comment:
why is it "professional" roles :sweat_smile: why not just roleAttributes/roles/roleNames etc

Reply:
Adding professional may give clarity I think. For example the user has the profession of teacher.
Also mentioned in the ticket.
https://intension-de.atlassian.net/browse/FWUV-236